packages/execution-environments/src/common/BaseSnapExecutor.test.ts

```diff
@@ -759,4 +759,351 @@ describe('BaseSnapExecutor', () => {
+          case 'first':
+            promise = wallet.request({ method: 'eth_blockNumber', params: [] })
+              .then(() => console.log('Jailbreak'));
+            return 'FIRST OK';
+          case 'second':
+            const timeout = new Promise((r) => setTimeout(() => r('TIMEOUT_RESOLVED_FROM_SECOND_CALL'), 1000));
+            return Promise.race([timeout, promise.then(() => 'SECOND OK')]);
+        }
+      }
+    `;
+    const executor = new TestSnapExecutor();
+
+    // --- Execute Snap
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'executeSnap',
+      params: [FAKE_SNAP_NAME, CODE, TIMER_ENDOWMENTS],
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    // --- Call Snap RPC
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 2,
+      method: 'snapRpc',
+      params: [
+        FAKE_SNAP_NAME,
+        FAKE_ORIGIN,
+        { jsonrpc: '2.0', method: 'first', params: [] },
+      ],
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      method: 'OutboundRequest',
+    });
+
+    const providerRequest = await executor.readRpc();
+    expect(providerRequest).toStrictEqual({
+      name: 'metamask-provider',
+      data: {
+        id: expect.any(Number),
+        jsonrpc: '2.0',
+        method: 'metamask_getProviderState',
+        params: undefined,
+      },
+    });
+
+    await executor.writeRpc({
+      name: 'metamask-provider',
+      data: {
+        jsonrpc: '2.0',
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        id: providerRequest.data.id!,
+        result: { isUnlocked: false, accounts: [] },
+      },
+    });
+
+    const blockNumRequest = await executor.readRpc();
+    expect(blockNumRequest).toStrictEqual({
+      name: 'metamask-provider',
+      data: {
+        id: expect.any(Number),
+        jsonrpc: '2.0',
+        method: 'eth_blockNumber',
+        params: [],
+      },
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      id: 2,
+      jsonrpc: '2.0',
+      result: 'FIRST OK',
+    });
+
+    // --- Call Snap RPC for the second time
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 2,
+      method: 'snapRpc',
+      params: [
+        FAKE_SNAP_NAME,
+        FAKE_ORIGIN,
+        { jsonrpc: '2.0', method: 'second', params: [] },
+      ],
+    });
+
+    await executor.writeRpc({
+      name: 'metamask-provider',
+      data: {
+        jsonrpc: '2.0',
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        id: blockNumRequest.data.id!,
+        result: '0xa70e77',
+      },
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      id: 2,
+      jsonrpc: '2.0',
+      result: 'TIMEOUT_RESOLVED_FROM_SECOND_CALL',
+    });
+
+    expect(consoleLogSpy).not.toHaveBeenCalledWith('Jailbreak');
```

Comment:
I cannot because there are two console log calls happening before the one I'm testing.
```
  console.log
    starting snap 'local:foo' in worker
  console.log
    Worker: Registering RPC message handler [AsyncFunction (anonymous)]
```
The first one for example is:
```typescript
  protected async startSnap(
    snapName: string,
    sourceCode: string,
    _endowments?: Endowments,
  ): Promise<void> {
    console.log(`starting snap '${snapName}' in worker`);
    // .........................
}
```
Another option was adding `toHaveBeenCalledTimes(3)`, but that looked even more confusing.